rescrape.py

```diff
@@ -36,92 +46,117 @@
+            "Try typing in the specific physical address using method above: "
+        )
         print("Trying to GeoCode the new address: ", Address)
     else:
-        print('The address has been GeoCoded!')
+        print("The address has been GeoCoded!")
         break
-
 # The search box coordinates are obtained via a geodesic measurement from the origin (e.g. user supplied address) to the
 # direct North, direct East, direct South, and direct West using the user supplied SearchBoxHalfWidth.
 # The North Boundary thus becomes the latitude of the geodesic measurement from the origin, and so on.
-NorthDestination = geodesic(kilometers=(SearchBoxHalfWidth * 1.60934)).destination(origin, 0.0)
+NorthDestination = geodesic(kilometers=(SearchBoxHalfWidth * 1.60934)).destination(
+    origin, 0.0
+)
 NorthBoundary = NorthDestination.latitude
-EastDestination = geodesic(kilometers=(SearchBoxHalfWidth * 1.60934)).destination(origin, 90.0)
+EastDestination = geodesic(kilometers=(SearchBoxHalfWidth * 1.60934)).destination(
+    origin, 90.0
+)
 EastBoundary = EastDestination.longitude
-SouthDestination = geodesic(kilometers=(SearchBoxHalfWidth * 1.60934)).destination(origin, 180.0)
+SouthDestination = geodesic(kilometers=(SearchBoxHalfWidth * 1.60934)).destination(
+    origin, 180.0
+)
 SouthBoundary = SouthDestination.latitude
-WestDestination = geodesic(kilometers=(SearchBoxHalfWidth * 1.60934)).destination(origin, 270.0)
+WestDestination = geodesic(kilometers=(SearchBoxHalfWidth * 1.60934)).destination(
+    origin, 270.0
+)
 WestBoundary = WestDestination.longitude
-print('Searching within a square area of half-width', SearchBoxHalfWidth, 'miles.')
+print("Searching within a square area of half-width", SearchBoxHalfWidth, "miles.")
 
 # Construct the Zillow Recently Sold Homes URL using the search boundary found above.
-ZillowURL = ['https://www.zillow.com/homes/recently_sold/?searchQueryState={'
-             '"pagination":{},'
-             '"mapBounds":{'
-             '"west":' + str(WestBoundary) +
-             ',"east":' + str(EastBoundary) +
-             ',"south":' + str(SouthBoundary) +
-             ',"north":' + str(NorthBoundary) + '},'
-             '"isMapVisible":true,'
-             '"mapZoom":8,'
-             '"filterState":{'
-             '"isForSaleByAgent":{"value":false},'
-             '"isForSaleByOwner":{"value":false},'
-             '"isNewConstruction":{"value":false},'
-             '"isForSaleForeclosure":{"value":false},'
-             '"isComingSoon":{"value":false},'
-             '"isAuction":{"value":false},'
-             '"isPreMarketForeclosure":{"value":false},'
-             '"isPreMarketPreForeclosure":{"value":false},'
-             '"isMakeMeMove":{"value":false},'
-             '"isRecentlySold":{"value":true}},'
-             '"isListVisible":true}']
-
-# Open up the Zillow web page to view the search box, results, and see how many pages there are.
-webbrowser.open_new(ZillowURL[0])

```

Comment:
I want this to be left in the code. I like how it opens up the main Zillow search page so the user can visualize what they are searching on. But more importantly, when Zillow changes their HTML, and I need to get to the website quickly to determine what the new HTML is for scraping purposes, having the exact web page already open allows me to quickly view the updated HTML that I will end up using in the code.

Reply:
As a matter of fact, disregard. Don't need to open the web page. Can simply just print it to screen so if the user wants to view it, they can click the link that comes up (or c&P it into their own browser).

Reply:
Disregard again. Printing URL to screen only hyperlinks part of it because there are "{" characters in it. So pycharm doesn't recognize the parts behind the "{" as also part of the hyperlink. Which means it makes it 1.) look like shit, and 2.) unclickable. Therefore, will put webbrowser.open back in. I think its alright. Just minimize it if you don't want to see it. sheesh.

Reply:
> Disregard again. Printing URL to screen only hyperlinks part of it because there are "{" characters in it. So pycharm doesn't recognize the parts behind the "{" as also part of the hyperlink. Which means it makes it 1.) look like shit, and 2.) unclickable. Therefore, will put webbrowser.open back in. I think its alright. Just minimize it if you don't want to see it. sheesh.

From  [Python documentation for format string syntax](https://docs.python.org/library/string.html#formatstrings):

> Format strings contain “replacement fields” surrounded by curly braces `{}`. Anything that is not contained in braces is considered literal text, which is copied unchanged to the output. If you need to include a brace character in the literal text, it can be escaped by doubling: `{{` and `}}`.

Reply:
> I want this to be left in the code. I like how it opens up the main Zillow search page so the user can visualize what they are searching on. But more importantly, when Zillow changes their HTML, and I need to get to the website quickly to determine what the new HTML is for scraping purposes, having the exact web page already open allows me to quickly view the updated HTML that I will end up using in the code.

I think this is better left as a debugging/development tool...